### PR TITLE
feat(User): add `avatarDecorationData`

### PR DESCRIPTION
**Pull request by almeidx**
**Please describe the changes this PR makes and why it should be merged:**
Upstream:
- https://github.com/discord/discord-api-docs/pull/6464

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating

### CODE REVIEW

**Review by didinele — APPROVED** [review]

File: packages/rest/src/lib/CDN.ts [hunk 1]
```diff
@@ -108,8 +117,18 @@ export class CDN {
 		userId: string,
 		userAvatarDecoration: string,
 		options?: Readonly<BaseImageURLOptions>,
+	): string;
+
+	public avatarDecoration(
+		userIdOrAsset: string,
+		assetOrOptions?: Readonly<ImageURLOptions> | string,
+		options?: Readonly<BaseImageURLOptions>,
```
Comment (ckohen) [inline packages/rest/src/lib/CDN.ts:125]: Is there a reason you used `BaseImageURLOptions` in the implementation but `ImageURLOptions` in the overload?

Comment (almeidx) [reply packages/rest/src/lib/CDN.ts:125]: The new overload uses the `dynamicMakeURL()` method, which takes `ImageURLOptions`

Comment (ckohen) [reply packages/rest/src/lib/CDN.ts:125]: Both overloads use `ImageURLOptions`, this is the only place `BaseImageURLOptions` is used

**Review by ckohen — APPROVED** [review]

File: packages/rest/src/lib/CDN.ts [hunk 2]
```diff
@@ -97,9 +97,18 @@ export class CDN {
...
+	 *
+	 * @param asset - The avatar decoration hash
+	 * @param options - Optional options for the avatar decoration preset
+	 */
+	public avatarDecoration(asset: string, options?: Readonly<ImageURLOptions>): string;
+
 	/**
 	 * Generates a user avatar decoration URL.
 	 *
+	 * @deprecated This overload is deprecated. Pass an hash instead.
```
Comment (sdanialraza) [inline packages/rest/src/lib/CDN.ts:111]:
```suggestion
	 * @deprecated This overload is deprecated. Pass a hash instead.
```

**Comment by Jiralite** [comment]
In discord.js, should this be checked for in `equals()`?

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/typings/index.d.ts [hunk 3]
```diff
@@ -3317,7 +3322,9 @@ export class User extends Base {
 
   public accentColor: number | null | undefined;
   public avatar: string | null;
+  /** @deprecated Use {@link avatarDecorationData} instead */
```
Comment (Jiralite) [inline packages/discord.js/typings/index.d.ts:3325]:
```suggestion
  /** @deprecated Use {@link User.avatarDecorationData} instead */
```

**Pull request merged**